geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/vlm/RasterSource.scala

```diff
@@ -150,6 +256,131 @@ object RasterSource {
     SpatialTiledRasterLayer(zoom, contextRDD)
   }
 
+  def readOrderedToLayout(
+    sc: SparkContext,
+    layerType: String,
+    paths: java.util.ArrayList[java.util.HashMap[String, String]],
+    layoutType: GPSLayoutType,
+    targetCRS: String,
+    numPartitions: Integer,
+    resampleMethod: ResampleMethod,
+    readMethod: String
+  ): SpatialTiledRasterLayer = {
+    val scalaPaths: Seq[Seq[(String, String)]] = paths.asScala.toSeq.map { _.asScala.toSeq }
+
+    val partitions =
+      numPartitions match {
+        case i: Integer => Some(i.toInt)
+        case null => None
+      }
+
+    readOrderedToLayout(
+      sc,
+      layerType,
+      sc.parallelize(scalaPaths, partitions.getOrElse(sc.defaultParallelism)),
+      layoutType,
+      targetCRS,
+      resampleMethod,
+      readMethod
+    )
+  }
+
+  def readOrderedToLayout(
+    sc: SparkContext,
+    layerType: String,
+    rdd: RDD[Seq[(String, String)]],
+    layoutType: LayoutType,
+    targetCRS: String,
+    resampleMethod: ResampleMethod,
+    readMethod: String
+  ): SpatialTiledRasterLayer = {
+    val rasterSourcesRDD: RDD[Seq[(String, RasterSource)]] =
+      (readMethod match {
+        case GEOTRELLIS =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, new GeoTiffRasterSource(v): RasterSource)
+              }
+            }
+          }
+        case GDAL =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, GDALRasterSource(v): RasterSource)
+              }
+            }
+          }
+      }).cache()
+
+    val orderedRasterSourcesRDD: RDD[Seq[RasterSource]] =
+      rasterSourcesRDD.map { keyedSource: Seq[(String, RasterSource)] =>
+        keyedSource.sortBy { _._1 }.map { case (_, source) => source }
+      }
+
+    val reprojectedSourcesRDD: RDD[Seq[RasterSource]] =
+      targetCRS match {
+        case crs: String =>
+          orderedRasterSourcesRDD.map {
+            _.map { _.reproject(CRS.fromString(crs), resampleMethod) }
+          }
+        case null =>
+          orderedRasterSourcesRDD
+      }
+
+    val rasterSummary: RasterSummary =
```

Comment:
Now that you have `RasterSummary` is a good chance to use `estimatePartitionsNumber` if no user input to override it was given.

Reply:
:+1: